networkx/generators/classic.py

```diff
@@ -566,6 +620,91 @@ def empty_graph(n=0, create_using=None, default=Graph):
     return G
 
 
+@nx._dispatch(graphs=None)
+def full_rary_tree(r, n, create_using=None):
+    """Creates a full r-ary tree of `n` nodes.
+
+    Sometimes called a k-ary, n-ary, or m-ary tree.
+    "... all non-leaf nodes have exactly r children and all levels
+    are full except for some rightmost position of the bottom level
+    (if a leaf at the bottom level is missing, then so are all of the
+    leaves to its right." [1]_
+
+    Parameters
+    ----------
+    r : int
+        branching factor of the tree
+    n : int
+        Number of nodes in the tree
+    create_using : NetworkX graph constructor, optional (default=nx.Graph)
+       Graph type to create. If graph instance, then cleared before populated.
+
+    Returns
+    -------
+    G : networkx Graph
+        An r-ary tree with n nodes
+
+    References
+    ----------
+    .. [1] An introduction to data structures and algorithms,
+           James Andrew Storer,  Birkhauser Boston 2001, (page 225).
+    """
+    G = empty_graph(n, create_using)
+    G.add_edges_from(_tree_edges(n, r))
+    return G
+
+
+def kneser_graph(n, k):
+    """Returns the Kneser Graph with parameters $n$ and $k$.
+
+    The Kneser Graph is the graph whose vertices are the $k$-subsets of
+    range($n$), such that two vertices are adjacent if their corresponding sets
+    are disjoint.
+
+    Parameters
+    ----------
+        n: int
+        Total number of elements.
+
+        k: int
+        Size of the subsets.
+
+    Returns
+    -------
+    G : NetworkX Graph
+
+    Examples
+    --------
+    >>> G = nx.kneser_graph(5, 2)
+    >>> G.number_of_nodes()
+    10
+    >>> G.number_of_edges()
+    15
+    >>> nx.is_isomorphic(G, nx.petersen_graph())
+    True
+
+    """
+    if n <= 0:
+        raise NetworkXError("n should be greater than zero")
+    if k <= 0 or k > n:
+        raise NetworkXError("k should be greater than zero and smaller than n")
+
+    G = nx.Graph()
+    # Create all k-subsets of [0, 1, ..., n-1]
+    subsets = list(itertools.combinations(range(n), k))
+
+    if 2 * k > n:
+        G.add_nodes_from(subsets)
```

Comment:
If I get this right, whenever `2*k>n` there can't be any edges (because the two sets of nodes must be disjoint and have k elements each). But why arent we also adding the nodes for the small k cases? Are we ensured that no nodes are isolated when `2*k <= n`?

Reply:
That is a very good question! I think when the k is small, there won't be nodes that are completely isolated from the rest (I.e. no neighbours). There are definitely cases where the graph is disjoint, but atleast NetworkX is able to infer the nodes of the graph from the edges. See [here](https://mathworld.wolfram.com/KneserGraph.html) for some more examples of Kneser graphs.

Reply:
Nice -- that webpage says that they are connected when N>2K, and later says that N=2K gives ladder rung graphs (disconnected edges forming a perfect matching -- no isolated nodes and all nodes have degree 1). So that's why we can avoid adding nodes when 2*k>n.  :tada: